### PR TITLE
Add TravisCI config file for automated testing

### LINKED ISSUE #7: Enable automated testing
Automated testing should be performed for commits to the master branch, and pull requests. Automated testing for other branches could be nice too, but it's not essential.

I've used [Travis CI](https://travis-ci.org/) for automated testing in the past, and been happy. I'm not set on them, though.

When doing this, make sure to fill out the `[badges]` section in `Cargo.toml`. For more on that, see documentation for [the manifest format](https://doc.rust-lang.org/cargo/reference/manifest.html). Also, err on the side of being strict. Here's some checks that should be able to produce failures:

```bash
cargo check
cargo test
cargo fmt
cargo clippy
cargo doc
```

### PR DESCRIPTION
**Pull request by cshlatz**
### What's included
The .travis.yml file in this PR does the following:
- Installs all required dependencies (for now, this is just libdbus-1-dev)
- Runs against all three distribution Rust channels: stable, beta and nightly. Nightly is allowed to fail. Given the nature of this project, I think it's important that this is done.
- Caches all of the dependencies, which significantly speeds up tests after the caching has happened.
- Installs the rust components clippy and fmt for additional tests
- Runs the following tests: check, test, clippy, fmt and doc

### Setup
TravisCI setup is incredibly easy. Please use their docs. You'll set it up for your repository specifically. The only thing you'll need to do after setting up on Travis is make sure that the .travis.yml file exists. Note that it's a hidden file.

You'll obviously want to run some tests. I did so by committing some bogus code. It may be important to find an issue that'll only break nightly? That may be tough to pull off though.

### Questions
- Do we care about the order tests are run?
- Do we care that we aren't running these tests with the --verbose flag? I think it makes the Travis logs too busy personally, and if you want that information, just run the test locally with the flag.
- Is cargo doc necessary/useful?
- Explore further GitHub integration. Refuse merges until all tests are passed?
- Is TravisCI the best option for Rust? I don't know, I didn't test the alternatives. I'm just a lowly contributor ;)

Fixes #7

### CODE REVIEW

**Comment by Ichimonji10** [comment]
> Do we care about the order tests are run?

Nope. Any order is fine.

> Do we care that we aren't running these tests with the --verbose flag? I think it makes the Travis logs too busy personally, and if you want that information, just run the test locally with the flag.

I'm happy with what you've done. The `--verbose` flag can be added later if needed.

> Is cargo doc necessary/useful?

Good question. In the past, I've relied upon the documentation builders to check for syntax errors, like broken links. For example, this will produce a broken link in the rust docs:

```
/// Here's a [broken link].
///
/// [broken liiiiiiink]: https://example.com
```

But some quick manual testing shows that `cargo doc` doesn't throw errors for this stuff. In fact, I can't find *any* ways to intentionally make `cargo doc` fail. `cargo doc --help` doesn't offer any sort of "strict mode" flag. And what's the point of a test if you can't find any circumstances under which it will fail?

I'm happy to drop the test. Alternately, we can keep it and pass `--no-deps` to skip building documentation for dependencies, which should speed things up. What do you think?

> Explore further GitHub integration. Refuse merges until all tests are passed?

The way I've handled this in the past is to integrate GitHub with Travis CI such that the latter can stick notifications on pull request pages showing whether the tests for a PR pass or fail, but to always allow merges anyway. Sometimes, the build will fail for reasons other than the PR, or sometimes there's a compelling reason to merge breaking changes, and I want to give humans the flexibility to do seemingly bad things if the situation demands it.

> Is TravisCI the best option for Rust? I don't know, I didn't test the alternatives. I'm just a lowly contributor ;)

It's the only CI service I've ever used. I'm returning to it because it did the job. I'm open to alternatives. I've heard of AppVeyor and CircleCI. There's more boutique options too, like the CI integrated into sourcehut.

----

FYI: [closing issues using keywords](https://help.github.com/en/articles/closing-issues-using-keywords)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - rustup component add clippy-preview
+    - rustup component add rustfmt
+
+# Be thorough, test everything available
+script:
+    - cargo check
+    - cargo test
+    - cargo clippy
+    - cargo fmt
+    - cargo doc
```
Comment (Ichimonji10) [inline .travis.yml]: I think it's reasonable to either drop this or add `--no-deps`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+before_script:
+    - rustup component add clippy-preview
+    - rustup component add rustfmt
+
+# Be thorough, test everything available
+script:
+    - cargo check
+    - cargo test
+    - cargo clippy
+    - cargo fmt
```
Comment (Ichimonji10) [inline .travis.yml]: I want to fail the build if this command changes the source code in any way. I bet something clever like `cargo fmt && git ...` can do the trick.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    apt:
+        packages:
+            - libdbus-1-dev
+
+# Cache the dependencies so they are only recompiled if they or the compiler were upgradeed
+# This adds the following directories to the cache
+# $TRAVIS_HOME/.cache/sccache
+# $TRAVIS_HOME/.cargo/
+# $TRAV_SHOME/.rustup/
+# target
```
Comment (Ichimonji10) [inline .travis.yml]: Specific details like "which directories are cached" are likely to become outdated. Maybe drop these details, or link to https://docs.travis-ci.com/user/languages/rust/#dependency-management ?

File: Cargo.toml [hunk 4]
```diff
@@ -20,3 +20,6 @@ serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 textwrap = {version="0", features = ["term_size"]}
 xdg = "2"
+
+[badges]
+travis-ci = {repository = "...", branch = "master"}
```
Comment (Ichimonji10) [inline Cargo.toml]: I think this should be:

```toml
travis-ci = {repository = "Ichimonji10/killjoy"}
```

I'll enable Travis CI ASAP.

File: .travis.yml (hunk 1 above)
Comment (cshlatz) [reply .travis.yml]: In my opinion, drop until we figure out how it's actually properly used in this context.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        packages:
+            - libdbus-1-dev
+
+# Cache the dependencies so they are only recompiled if they or the compiler were upgraded
+# See https://docs.travis-ci.com/user/languages/rust/#dependency-management for more info.
+cache: cargo
+
+# Install clippy and fmt so they're available for the script
+before_script:
+    - rustup component add clippy-preview
```
Comment (Ichimonji10) [inline .travis.yml]: What's the difference between `clippy-preview` and `clippy`?

**Pull request merged**